test/e2e_yaml_test.go

```diff
@@ -0,0 +1,238 @@
+	err := c.ClusterTaskClient.Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func YamlSubtest(path string, validateFunc func(t *testing.T, c *clients, name string), kind string) func(subt *testing.T) {
+	return func(subt *testing.T) {
+		subt.Parallel()
+
+		// Setup unique namespaces for each test so they can run in complete
+		// isolation
+		c, namespace := setup(subt)
+
+		adminAvailable := true
+
+		// Same check as in github.com/tektoncd/plumbing/scripts/e2e-tests.sh
+		_, err := c.RbacV1Client.ClusterRoleBindings().Get("cluster-admin-binding", metav1.GetOptions{})
+		if err != nil {
+			subt.Logf("cluster-admin-binding result: %s", err)
+			adminAvailable = false
+		}
+
+		// If cluster-admin-binding is available, use it as CI is running, and
+		// admin is required for some examples
+		if adminAvailable {
+			subt.Log("Using cluster-admin user for YAML tests")
+			adminOverride := &clientcmd.ConfigOverrides{
+				Context: api.Context{
+					AuthInfo: "cluster-admin",
+				},
+			}
+
+			c, namespace = setupWithOverrides(subt, adminOverride)
+		}
+
+		knativetest.CleanupOnInterrupt(func() { tearDown(subt, c, namespace) }, subt.Logf)
+		defer tearDown(subt, c, namespace)
+
+		inputExample, err := ioutil.ReadFile(path)
+
+		if err != nil {
+			subt.Fatal(err)
+		}
+
+		subbedInput, err := SubEnv(inputExample, namespace)
+		if err != nil {
+			subt.Fatal(err)
+		}
+
+		out, err := KoCreate(subbedInput, namespace)
+		if err != nil {
+			subt.Fatalf("%s Output: %s", err, out)
+		}
+
+		// Parse from KoCreate for now
+		name, err := GetCreatedTektonCrd(out, kind)
+		if name == "" {
+			// Nothing to check from ko create, this is not a taskrun or pipeline
+			// run
+			subt.Skipf("pipelinerun or taskrun not created")
+		} else if err != nil {
+			subt.Fatal(err)
+		}
+
+		// NOTE: If an example creates more than one clustertask, they will not all
+		// be cleaned up
+		clustertask, err := GetCreatedTektonCrd(out, "clustertask")
+		if clustertask != "" {
+			knativetest.CleanupOnInterrupt(func() { DeleteClusterTask(subt, c, clustertask) }, subt.Logf)
+			defer DeleteClusterTask(subt, c, clustertask)
+		} else if err != nil {
+			subt.Fatal(err)
+		}
+
+		validateFunc(subt, c, name)
+	}
+}
+
+func GetExamplePaths(t *testing.T, dir string) []string {
+	var examplePaths []string
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Do not append root and any other folders named "examples"
+		if info.Name() == "examples" && info.IsDir() {
+			return nil
+		}
+		if info.Name() == "no-ci" && info.IsDir() {
+			return filepath.SkipDir
+		}
+		if info.IsDir() == false && filepath.Ext(info.Name()) == ".yaml" {
+			examplePaths = append(examplePaths, path)
+			return nil
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return examplePaths
+}
+
+func ExtractTestName(baseDir string, path string) string {
```

Comment:
if we're making these exported functions it'd be great to get some docstrings on them